site/docs/v1/tech/themes/kickstart-custom-theme.adoc

```diff
@@ -0,0 +1,400 @@
+
+```bash
+npm install
+npm start
+```
+
+Open an incognito window and visit `http://localhost:3000`.
+
+You will be taken to the same landing page that you saw in the 5-minute guide. However, when you click "Login" this time, you will see your custom theme applied to the OAuth authorize page.
+
+image::themes/kickstart-theme/dark-authorize.png[Dark theme applied,width=1200, role=bottom-cropped]
+
+Enter the same credentials you used to login to the admin panel and click "Logout" to see the OAuth logout page.
+
+
+== Modifying the Default Messages
+
+Let's take it one step further and assume we want to change the content of some of the messages on the OAuth pages. For example, consider the "Forgot your password?" message, which shows up on the OAuth authorize page.
+
+image::themes/kickstart-theme/forgot-password.png[Forgot password,width=1200, role=bottom-cropped]
+
+Let's say we want to change this to say "Forgot your password? Click here." We can do this by adding a `defaultMessages` property to `json/updateTheme.json`.
+
+The `defaultMessages` string requires at least all of the messages defined in the FusionAuth default shipped messages file to be present, as it updates all messages as a single unit. The easiest way to accomplish this is to create a new file called `defaultMessages.txt` in your `kickstart` folder and copy-paste these messages into it.
+
+The messages can be accessed by editing your custom theme, navigating to the "Messages" page, and clicking the "Edit" button.
+
+image::themes/kickstart-theme/messages.png[Theme messages,width=1200]
+
+Copy the entire contents of that box into your `defaultMessages.txt` file, find the `forgot-your-password` message (line 65), and modify it to "Forgot your password? Click here."
+
+```json
+{
+  "method" : "PATCH",
+  "url" : "api/theme/#{themeID}",
+  "body" : {
+    "theme" : {
+      "stylesheet" : "@{darkTheme.css}",
+      "defaultMessages" : "@{defaultMessages.txt}"
+    }
+  }
+}
+````
+```
+[NOTE]
+====
+Kickstart will not run if it sees any users, API keys, or applications in the FusionAuth database. This is to prevent data loss. If you can login to the FusionAuth administrative user interface, Kickstart will not run.
+====
+
+Once you have modified `updateTheme.json`, you will need to clear the volumes created when you launched the FusionAuth instance to allow the Kickstart to rerun. When you relaunch the instance using the `docker-compose up` command and apply the dark theme, the result will be as below with the updated message:
+
+
+
+image::themes/kickstart-theme/updated-message.png[Updated message,width=1200, role=bottom-cropped]
+
+== Conclusion
+
+This guide has shown you how to use Kickstart to launch a reproducible FusionAuth instance with a custom theme. The complete set of files for this project can be found https://github.com/ritza-co/fusion-kickstart-custom-theme[here].
```

Comment:
Will need to pull this over from ritza. In the future, I can also create you a repo and grant you access if that's easier.

Reply:
probably still easiest if you just fork it and then we can update the link (or even put them in ahead of time as we'll always be able to predict the URL) We're also happy to push up the same work to a repo you create and give us access to if you prefer (then you won’t see the “forked from” in the FusionAuth version).  If you want to create them your side, can you give access to me @sixhobbits and Tatenda @rideam

Reply:
Works for me. I end up pulling it down and repushing anyway with a different origin, so as to avoid the `forked` message.

But that's not too much work.

Reply:
ok perfect, https://github.com/fusionAuth/fusion-kickstart-custom-theme is 404 for me though, did you put it somewhere else? Or maybe it's private?

Reply:
Here you are: https://github.com/FusionAuth/fusionauth-example-kickstart-theme .